build-setup/linux.md

```diff
@@ -407,25 +407,23 @@ Building IPEX requires a lot of memory. To reduce this requirement, we can set t
 export MAX_JOBS=1
 ```
 
-IPEX expects that the `blas-devel` library package be installed:
-```bash
-yum install blas-devel.x86_64
-```
+The IPEX third party library dependency `LIBXSMM` link stage has a dependency on `libblas`. This dependency can be removed by setting `BLAS=0` in the environment.
```

Comment:
Please can you confirm with `ldd` that the `BLAS=0` workaround works. Also check if there are any other niche OS library dependencies of either IPEX or PyTorch 2.1.

I am a bit worried about the side effects of setting `BLAS=0` for `LIBXSMM`. It might mean that then certain operations of `LIBXSMM` simply don't work. We'd only detect this at runtime when a model that uses IPEX needs to do one of these operations. Our PR build tests are nowhere near up to the job of detecting subtle problems like this.

For 8.12 this doesn't actually matter, as we're not using IPEX. However, this is an extra piece of work to do before we ever do use in in the future.

This line in the `LIBXSMM` `Makefile` makes me wonder if it's possible to force the BLAS to always be MKL, and then we could set the build up like that: https://github.com/libxsmm/libxsmm/blob/54b8bb32042d204ff6fe550fa70d961f22c0b99e/Makefile#L246

Please can you add a TODO in this file to revisit `BLAS=0` before actually using IPEX in production, and also open an issue to investigate the best way to build IPEX that's both portable to all supported Linux distributions _and_ means that IPEX won't have any missing areas of functionality that cause runtime errors.

But don't try and solve this as part of this PR.

Reply:
Those dynamic dependencies you've listed look good.

The BLAS documentation for `LIBXSMM` isn't 100% clear to me, and then there's the extra complexity on top of when/how IPEX uses `LIBXSMM`.

Thanks for opening #2589. Nothing more is needed for this PR though.